application-analytics-ui/src/main/resources/Analytics/Code/Macros/VelocityMacros.xml

```diff
@@ -61,5 +61,104 @@
+&lt;div class="analyticsActions"&gt;
+  &lt;div class="analyticsRowEvolution"  data-parameter=$escapetool.url($parameter)&gt;
+    &lt;a class="action" title="$escapetool.xml($services.localization.render(
+      'analytics.mostViwedMacros.action.rowEvolution'))" href="#"&gt;
+      &lt;span class="action-icon"&gt;$services.icon.renderHTML('eye')&lt;/span&gt;&lt;span class="action-label"&gt;
+      $escapetool.xml($services.localization.render('analytics.action.label.rowEvolution'))&lt;/span&gt;
+    &lt;/a&gt;
+  &lt;/div&gt;
+&lt;/div&gt;
+#end
+
+#macro(durationSelect $className $options)
+&lt;select class="$className form-control"&gt;
+  #foreach($option in $options)
+    &lt;option value="$option"&gt;$option&lt;/option&gt;
+  #end
+&lt;/select&gt;
+#end
+
+
+#macro(durationOption $value)
+  &lt;option value="$value"&gt;
+    $services.localization.render('analytics.rowEvolution.modal.' + $value.toLowerCase() + '.label')
+  &lt;/option&gt;
+#end
+
+#macro(durationSelectList $class $options)
+  &lt;select class="$class form-control"&gt;
+    #foreach($option in $options)
+      &lt;option value="$option"&gt;$option&lt;/option&gt;
+    #end
+  &lt;/select&gt;
+#end
+
+#macro(fieldsSelect $fields)
+  &lt;select class="form-control analyticsApplicationRowEvolutionField"&gt;
+    #foreach($key in $fields.keySet())
+      &lt;option value="$fields.get($key)"&gt;
+        $services.localization.render('analytics.rowEvolution.field.' + $key)
+      &lt;/option&gt;
+    #end
+  &lt;/select&gt;
+#end
+
+#macro(modalRowEvolution $fields $requestPageName)
+#set ($requestPageURL = $xwiki.getURL($requestPageName))
+{{html clean=false}}
+&lt;div class="modal fade analyticsRowEvolutionModal" tabindex="-1" role="dialog" aria-labelledby="chartModalLabel"
+     aria-hidden="true"&gt;
+  &lt;div class="modal-dialog modal-lg"&gt;
+    &lt;div class="modal-content"&gt;
+      &lt;div class="modal-header"&gt;
+        &lt;button type="button" class="close" data-dismiss="modal" aria-label="Close"&gt;
+          &lt;span aria-hidden="true"&gt;&amp;times;&lt;/span&gt;
+        &lt;/button&gt;
+        &lt;h4 class="modal-title" class="chartModalLabel"&gt;
+          $services.localization.render('analytics.rowEvolution.label')
+        &lt;/h4&gt;
+      &lt;/div&gt;
+      &lt;div class="modal-body"&gt;
+        &lt;label for="analyticsApplicationModalPeriodSelect"&gt;
+          $services.localization.render('analytics.rowEvolution.period')
+        &lt;/label&gt;
+        &lt;select class="analyticsApplicationModalPeriodSelect form-control"&gt;
+          #durationOption("Day")
+          #durationOption("Week")
+          #durationOption("Month")
+          #durationOption("Year")
+        &lt;/select&gt;
+        &lt;div class="analyticsRowEvolutionOptions" data-request-page=$requestPageURL data-period='day' data-date='30'&gt;
+          #durationSelectList("analyticsApplicationRowEvolutionDurationDay" [30, 8, 60, 90, 180, 365, 500])
```

Comment:
Here as well, you don't need to repeat the `analyticsApplication` sufix. Maybe `rowEvolutionPeriodDayLength`, `rowEvolutionPeriodMonthLength`, ...? Or `chartPeriodDayLength`